src/module-server/transform-imports.ts

```diff
@@ -51,8 +55,32 @@ export const transformImports = async (
   id: string,
   { resolveImportMeta, resolveId, resolveDynamicImport }: Options = {},
 ) => {
-  // eslint-disable-next-line @cloudfour/typescript-eslint/await-thenable
-  const [imports] = await parse(code, id);
+  let imports;
+  try {
+    // eslint-disable-next-line @cloudfour/typescript-eslint/await-thenable
+    imports = (await parse(code, id))[0];
+  } catch (error) {
+    if (!('idx' in error)) throw error;
+    const linesUntilError = code.slice(0, error.idx).split('\n');
+    const line = linesUntilError.length;
+    const column = linesUntilError[linesUntilError.length - 1].length;
+    const frame = createCodeFrame(code, line - 1, column);
+    let message = `${colors.red(colors.bold(error.message))}
+
+${colors.red(`${id}:${line}:${column + 1}`)}
+
+${frame}
+`;
+    if (!jsExts.test(id) && !cssExts.test(id))
+      message += `${colors.yellow(
+        'You may need to add transform plugins to handle non-JS input',
```

Comment:
Nit: I'm not sure I would understand this error if I ran into it while using Pleasantest. Is there any additional context we could add to make this clearer?

Reply:
How does this sound: `You may need to add plugins to the module server to handle <extension> files`

Reply:
This error _only_ appears if the extension is not .js/.cjs/.mjs/.ts/.jsx/.tsx/.css/.sass/.scss/.less/.styl so I don't think it will be very common. I think that adding the file extension would make it more helpful because then people will hopefully understand why the file can't be parsed (i.e. if someone tries to load a .coffee file 😂)

Reply:
> How does this sound: `You may need to add plugins to the module server to handle <extension> files`

I feel this is more helpful, what do you think, @Paul-Hebert? 🙂

Reply:
Me too!